### PR TITLE
Removing list "Lifecycle Events"

### PR DESCRIPTION
**Pull request by ThomasLandauer**
As announced in https://github.com/doctrine/orm/pull/9184#issuecomment-965837780

I'm committing the events one by one, to make it easier to follow.

Notes:
* postUpdate, postRemove, postPersist aren't similar enough to have them in one section, so it might be better to separate them.
* This warning I don't understand. I moved it to `postRemove`. Should it be repeated on all "events triggered after an entity removal" (which ones?):
    > The ``postRemove`` event or any events triggered after an entity removal
    can receive an uninitializable proxy in case you have configured an entity to
    cascade remove relations. In this case, you should load yourself the proxy in
    the associated ``pre*`` event.

### CODE REVIEW

File: docs/en/reference/events.rst [hunk 1]
```diff
@@ -187,94 +189,6 @@ several reasons:
...
-The ``EventArgs`` instance received by the listener gives access to the entity,
-``EntityManager`` instance and other relevant data.
-
-.. note::
-
-    All Lifecycle events that happen during the ``flush()`` of
-    an ``EntityManager`` have very specific constraints on the allowed
-    operations that can be executed. Please read the
-    :ref:`reference-events-implementing-listeners` section very carefully
-    to understand which operations are allowed in which lifecycle event.
```
Comment (SenseException) [inline docs/en/reference/events.rst:276]: I think this note has a history and should be kept. Maybe below the table?

**Review by SenseException — CHANGES_REQUESTED** [review]

File: docs/en/reference/events.rst (hunk 1 above)
Comment (ThomasLandauer) [reply docs/en/reference/events.rst:276]: Well, this entire page is about documenting which actions are allowed where. So adding a note like "Not everything is allowed everywhere." doesn't add any value IMO. Was the history that some users didn't understand what is allowed where? Then the solution is not to add an *unspecific* "Watch out, guys!" warning, but rather to exactly list all the constraints for each event. => And that's what I'm trying ;-)
Bottom line: If you really insist, I'll bring it back :-)

Comment (SenseException) [reply docs/en/reference/events.rst:276]: It's a note with a reference to another part of the docs, so I think you have a point.

**Comment by SenseException** [comment]
Great work. Please change your commit messages into meaningful ones as described in https://www.doctrine-project.org/contribute/index.html#crafting-meaningful-commit-messages.

**Comment by derrabus** [comment]
We can (and should) squash the commits on merge, @SenseException.

**Comment by ThomasLandauer** [comment]
If you squash them, should I still change something?

**Review by derrabus — APPROVED** [review]
I think, we're good. Thank you.
